pkg/snapshotmgr/snapshot_manager.go

```diff
@@ -244,65 +253,98 @@ func (this *SnapshotManager) createSnapshot(peID astrolabe.ProtectedEntityID, ta
 		return updatedPeID, nil
 	}
 
+	_, err = this.UploadSnapshot(peID, updatedPeID, peSnapID, pe, ctx, backupRepositoryName, snapshotRef)
+	if err != nil {
+		return astrolabe.ProtectedEntityID{}, err
+	}
+
+	return updatedPeID, nil
+}
+
+/*
+Creates an Upload CR
+*/
+func (this *SnapshotManager) UploadSnapshot(peID astrolabe.ProtectedEntityID, updatedPeID astrolabe.ProtectedEntityID,
+	peSnapID astrolabe.ProtectedEntitySnapshotID, pe astrolabe.ProtectedEntity, ctx context.Context, backupRepositoryName string,
+	snapshotRef string) (*v1api.Upload, error) {
 	this.Info("Start creating Upload CR")
 	config, err := rest.InClusterConfig()
 	if err != nil {
 		this.WithError(err).Errorf("Failed to get k8s inClusterConfig")
-		return updatedPeID, err
+		return nil, err
 	}
 	pluginClient, err := plugin_clientset.NewForConfig(config)
 	if err != nil {
 		this.WithError(err).Errorf("Failed to get k8s clientset from the given config: %v ", config)
-		return updatedPeID, err
+		return nil, err
 	}
 
 	// look up velero namespace from the env variable in container
 	veleroNs, exist := os.LookupEnv("VELERO_NAMESPACE")
 	if !exist {
 		this.WithError(err).Errorf("CreateSnapshot: Failed to lookup the env variable for velero namespace")
-		return updatedPeID, err
+		return nil, err
 	}
 
-	uploadBuilder := builder.ForUpload(veleroNs, "upload-"+peSnapID.GetID()).
+	uploadUUID, err := uuid.NewRandom()
+	if err != nil {
+		this.WithError(err).Errorf("Could not create upload UUID")
+		return nil, err
+	}
+	uploadName := "upload-" + uploadUUID.String()
```

Comment:
The expectation is the upload name is "upload-"+snap-id, the name format is used to look up the upload CRs during backup delete for cancellation.

Reply:
I'm not sure if we should rely on upload name to find the snapshot id.  What if there are two backups happening at the same time for the same snapshot id?  I know that we don't support concurrent backups yet, but in the future we may.  We should probably take another look and see if there's another way to find this correlation.

Reply:
Actually, the other way round, we use the snap-id that comes in through delete backup requests to search for outstanding uploads. There is pre-existing code in delete snapshot that expects the upload CR name in this format, current change causes regression, you could revert back to the original naming format and open a jira ticket to track the issue.